src/nvim/ex_eval.c

```diff
@@ -68,7 +68,16 @@
 #define THROW_ON_INTERRUPT true
 #define THROW_ON_INTERRUPT_TRUE
 
-#define discard_pending_return(p) tv_free((typval_T *)(p))
+static inline void discard_pending_return(void *p)
+  REAL_FATTR_ALWAYS_INLINE REAL_FATTR_PURE;
```

Comment:
Must not use REAL_FATTR in `*.c` files unless you really have to. Fixtures in test is the case, files in src are almost never. Remove the declaration completely and use FUNC_ATTR after declaration part of function definition.

Reply:
This is by no means PURE function, pure means function without side-effects except for accessing global memory (CONST if it also does not access read-write global memory (though compiler docs say that it should not access *any* global memory)) and you can *never* have a useful function which is both pure and having `void` return.